src/sagemaker/jumpstart/curated_hub/accessors/public_hub_s3_accessor.py

```diff
@@ -0,0 +1,124 @@
+        self._studio_metadata_map = get_studio_model_metadata_map_from_region(
+            region
+        )  # Necessary for SDK - Studio metadata drift
+
+    def get_bucket(self) -> str:
+        """Retrieves s3 bucket"""
+        return self._bucket
+
+    def get_inference_artifact_s3_reference(
+        self, model_specs: JumpStartModelSpecs
+    ) -> S3ObjectReference:
+        """Retrieves s3 reference for model inference artifact"""
+        return create_s3_object_reference_from_uri(
+            self._jumpstart_artifact_s3_uri(JumpStartScriptScope.INFERENCE, model_specs)
+        )
+
+    def get_training_artifact_s3_reference(
+        self, model_specs: JumpStartModelSpecs
+    ) -> S3ObjectReference:
+        """Retrieves s3 reference for model training artifact"""
+        return create_s3_object_reference_from_uri(
+            self._jumpstart_artifact_s3_uri(JumpStartScriptScope.TRAINING, model_specs)
+        )
+
+    def get_inference_script_s3_reference(
+        self, model_specs: JumpStartModelSpecs
+    ) -> S3ObjectReference:
+        """Retrieves s3 reference for model inference script"""
+        return create_s3_object_reference_from_uri(
+            self._jumpstart_script_s3_uri(JumpStartScriptScope.INFERENCE, model_specs)
+        )
+
+    def get_training_script_s3_reference(
+        self, model_specs: JumpStartModelSpecs
+    ) -> S3ObjectReference:
+        """Retrieves s3 reference for model training script"""
+        return create_s3_object_reference_from_uri(
+            self._jumpstart_script_s3_uri(JumpStartScriptScope.TRAINING, model_specs)
+        )
+
+    def get_default_training_dataset_s3_reference(
+        self, model_specs: JumpStartModelSpecs
+    ) -> S3ObjectReference:
+        """Retrieves s3 reference for s3 directory containing model training datasets"""
+        return create_s3_object_reference_from_bucket_and_key(
+            self.get_bucket(), self._get_training_dataset_prefix(model_specs)
+        )
+
+    def _get_training_dataset_prefix(self, model_specs: JumpStartModelSpecs) -> str:
+        """Retrieves training dataset location"""
+        studio_model_metadata = self._studio_metadata_map[model_specs.model_id]
+        return studio_model_metadata["defaultDataKey"]
+
+    def get_demo_notebook_s3_reference(self, model_specs: JumpStartModelSpecs) -> S3ObjectReference:
+        """Retrieves s3 reference for model demo jupyter notebook"""
+        framework = get_model_framework(model_specs)
+        key = f"{framework}-notebooks/{model_specs.model_id}-inference.ipynb"
+        return create_s3_object_reference_from_bucket_and_key(self.get_bucket(), key)
+
+    def get_markdown_s3_reference(self, model_specs: JumpStartModelSpecs) -> S3ObjectReference:
+        """Retrieves s3 reference for model markdown"""
+        framework = get_model_framework(model_specs)
+        key = f"{framework}-metadata/{model_specs.model_id}.md"
```

Comment:
We should consider updating the Studio metadata to have this info explicitly. That will help with overall system flexibility here and in Studio (rather than relying on a very specific naming convention). Not a blocker right now though.